docs/architecture/ADR-002-deposit-assets.md

```diff
@@ -1,129 +1,60 @@
-
-MsgDepositAsset must use only allow-listed, non-uToken denominations. MsgWithdrawAsset must use only uToken denominations.
-
-These messages should trigger the appropriate reaction (disbursement of uTokens after deposit, return of assets on withdrawal). The exchange rate defined in ADR-001 must be used.
-
-_Note: The `Coin` type seen in the `Amount` fields contains a single token denomination and amount._
-
-It is necessary that `MsgSupply` and `MsgWithdrawAsset` be signed by the supplier's account. According to the [Transactions Page](https://docs.cosmos.network/master/core/transactions.html)
-
-> Every message in a transaction must be signed by the addresses specified by its GetSigners.
-
-Thus `MsgSupply.GetSigners` and `MsgWithdrawAsset.GetSigners` should return the `Supplier` address.
+When a user withdraws tokens, the tokens are moved from the `x/leverage` module account back to their wallet. Simultaneously, uTokens are removed from the user's wallet and burned by the module. The module uses its current token:uToken exchange rate.
 
-### API
-
-Both CLI and gRPC must be supported when sending the above message types, and all necessary handlers must be created in order to process and validate them as transactions. As part of this initial feature, an exact list of such steps required when adding message types will be created and added to future issues.
-
-### Testing
-
-Assuming a placeholder token allow-list of one element (e.g. `umee`), and a uToken existing (e.g. `u-umee`), an end-to-end test can be created in which one user account sends a `MsgSupply` and a `MsgWithdrawAsset` of the appropriate token types.
-
-## Considerations
-
-- IBC tokens coming from different channels will have different Denom. Hence we have a fragmentation problem. Example: `atom` coming directly from the Cosmos Hub will have different IBC denom than an `atom` coming directly from Osmosis.
+If the full requested amount of tokens is not available for withdrawal, the transaction fails.
 
 ## Consequences
 
 ### Positive
-
-- Banking module already provides underlying functionality
-- Asset deposit does not require us to modify state beyond what is already done by the `x/bank` module (e.g. account balances): deposit of an assets into a module account is atomic with the minting of respective uTokens to the sender's account, and uToken ownership is the sole requirement for asset withdrawal. We do not need to track individual user deposit history.
+- [x/bank module]([Cosmos Bank Keeper](https://github.com/cosmos/cosmos-sdk/blob/v0.44.0/x/bank/spec/02_keepers.md)) provides the underlying functionality.
 
 ### Negative
+- If tokens were to be sent unexpectedly to the `x/leverage` module account (see the warning on [this page](https://docs.cosmos.network/master/modules/bank/)), it would have the effect of donating said tokens to existing suppliers if the tokens were an accepted type. Otherwise, the tokens would remain inert.
```

Comment:
```suggestion
- If tokens were to be sent unexpectedly to the `x/leverage` module account (see the warning on [this page](https://docs.cosmos.network/master/modules/bank/)), it would have the effect of donating said tokens to existing suppliers if the tokens were an accepted type. Otherwise, the tokens would remain inert.
- uToken value is not the same as the underlying Token - can create some confusions
- need to involve extra calculation (exchange rates) for all uToken operations
```

Reply:
In general, our approach to ADRs has been
- Create them one at a time, recording our reasoning as design decisions are made
- Edit later to improve the text, but not substantially alter the decisions themselves
- Create separate ADR if a major change to the design is decided much later (e.g. `Adjusted Borrow * Interest Scalar` replacing `Borrow` & `InterestEpoch`)

Contrast with the spec, which is considered a single document describing the system as it currently stands.